src/Microsoft.AspNetCore.WebUtilities/WebEncoders.cs

```diff
@@ -124,38 +216,104 @@ public static string Base64UrlEncode(byte[] input, int offset, int count)
                 return string.Empty;
             }
 
-            // We're going to use base64url encoding with no padding characters.
-            // See RFC 4648, Sec. 5.
-            char[] buffer = new char[GetNumBase64CharsRequiredForInput(count)];
-            int numBase64Chars = Convert.ToBase64CharArray(input, offset, count, buffer, 0);
+            char[] buffer = new char[GetArraySizeRequiredToEncode(count)];
+            int numBase64Chars = Base64UrlEncode(input, offset, count, buffer, outputOffset: 0);
+
+            return new String(buffer, startIndex: 0, length: numBase64Chars);
+        }
 
-            // Fix up '+' -> '-' and '/' -> '_'
-            for (int i = 0; i < numBase64Chars; i++)
+        /// <summary>
+        /// Encodes <paramref name="input"/> using base64url encoding.
+        /// </summary>
+        /// <param name="input">The binary input to encode.</param>
+        /// <param name="offset">The offset into <paramref name="input"/> at which to begin encoding.</param>
+        /// <param name="count">The number of <c>byte</c>s from <paramref name="input"/> to encode.</param>
+        /// <param name="output">
+        /// Buffer to receive the base64url-encoded form of <paramref name="input"/>. Array must be large enough to
+        /// hold <paramref name="outputOffset"/> characters and the full base64-encoded form of
+        /// <paramref name="input"/>, including padding characters.
+        /// </param>
+        /// <param name="outputOffset">
+        /// The offset into <paramref name="output"/> at which to begin writing the base64url-encoded form of
+        /// <paramref name="input"/>.
+        /// </param>
+        /// <returns>
+        /// The number of characters written to <paramref name="output"/>, less any padding characters.
+        /// </returns>
+        public static int Base64UrlEncode(byte[] input, int offset, int count, char[] output, int outputOffset)
```

Comment:
By convention, the `count` in these methods always comes last (`Buffer.BlockCopy`, `string.Compare`)